packages/headless/src/controllers/facets/facet/headless-facet.ts

```diff
@@ -330,7 +338,7 @@ export function buildFacet(
       const request = getRequest();
       const response = getResponse();
 
-      const isLoading = engine.state.search.isLoading;
+      const isLoading = getIsLoading();
```

Comment:
Is this change needed or just an improvement you saw as you were working on this ?

If it's needed, I'd like to understand why ;)

Reply:
Yep, I made this change to resolve a typescript error. Inside getters only, typescript doesn't correctly infer the state of the engine. It sees it as `unknown`.

Everywhere else in the controller, ts understands the correct type via the typeguard.

```
buildSomeController(
engine: Engine<unknown>
) {
    // engine state is "unknown"
    if (!loadReducers(engine)) {
        throw Error
    }

    // if we are here, typescript understands that the engine state is that specified by the loadReducers type guard

    return {
        get state() {
           // here, typescript goes back to thinking the engine state is "unknown" for some reason.
       }
    }
}
```

Reply:
An alternative would be to do this, and update references to use `resolvedEngine` inside the controller:

```
export function buildFacet(engine: Engine<unknown>, props: FacetProps): Facet {
  if (!loadFacetReducers(engine)) {
    throw loadReducerError;
  }

  const resolvedEngine = engine;
```

Down-side is one would be able to use `engine` or `resolvedEngine`, except inside getters where only `resolvedEngine` would work.

Reply:
I created a report for the TS team: https://github.com/microsoft/TypeScript/issues/43676